control-plane/pkg/reconciler/channel/v2/channelv2.go

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/reconciler"
+
+	messagingv1beta1 "knative.dev/eventing-kafka/pkg/apis/messaging/v1beta1"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/kafka"
+
+	internals "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing"
+	internalscg "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1"
+	internalsclient "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/clientset/versioned"
+	internalslst "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/listers/eventing/v1alpha1"
+)
+
+const (
+	// TopicPrefix is the Kafka Channel topic prefix - (topic name: knative-messaging-kafka.<channel-namespace>.<channel-name>).
+	TopicPrefix          = "knative-messaging-kafka"
+	DefaultDeliveryOrder = internals.Ordered
+)
+
+type Reconciler struct {
+	ConsumerGroupLister internalslst.ConsumerGroupLister
+	InternalsClient     internalsclient.Interface
+}
+
+func (r *Reconciler) ReconcileKind(ctx context.Context, channel *messagingv1beta1.KafkaChannel) reconciler.Event {
```

Comment:
In this reconciler we need to handle the ingress part, are you planning on doing it in a different PR?

Reply:
The ingress part should be set like here https://github.com/knative-sandbox/eventing-kafka-broker/blob/e32734a003082847ed1d98a27f07f6a362e26248/control-plane/pkg/reconciler/channel/channel.go#L529-L555

Then for each subscriber like here: https://github.com/knative-sandbox/eventing-kafka-broker/blob/e32734a003082847ed1d98a27f07f6a362e26248/control-plane/pkg/reconciler/channel/channel.go#L406-L429 instead of this logic: https://github.com/knative-sandbox/eventing-kafka-broker/blob/e32734a003082847ed1d98a27f07f6a362e26248/control-plane/pkg/reconciler/channel/channel.go#L433-L451 we just create a new `ConsumerGroup` resource accordingly.

Reply:
Done